src/Connect/printer.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <cstdint>
+#include <cstddef>
+#include <tuple>
+
+namespace con {
+
+class Printer {
+public:
+    struct PrinterInfo {
+        static constexpr size_t SER_NUM_STR_LEN = 19;
+        static constexpr size_t SER_NUM_BUFR_LEN = (SER_NUM_STR_LEN + 1);
+
+        static constexpr size_t FINGERPRINT_SIZE = 50;
+        static constexpr size_t FINGERPRINT_BUFF_LEN = (FINGERPRINT_SIZE + 1);
+        static constexpr size_t FINGERPRINT_HDR_SIZE = 16;
+
+        bool appendix;
+        const char *firmware_version;
+        char serial_number[SER_NUM_BUFR_LEN];
+        char fingerprint[FINGERPRINT_BUFF_LEN];
+    };
+
+    enum class DeviceState {
+        Unknown,
+        Idle,
+        Printing,
+        Paused,
+        Finished,
+        Prepared,
```

Comment:
State "Prepared" was renamed to "Ready" in connect. Maybe we should do the same in FW to avoid confusion.

Reply:
I'm looking into the docs… and it can contain both `IDLE`, `READY` and `PREPARED`. I'm going to inquire about what each one of them mean and what the current status is before doing that and then somehow unify it.

Reply:
Done…